Donation/DonationAmountView.swift

```diff
@@ -0,0 +1,200 @@
+ * $Id$
+ *
+ * Authors: Carola Nitz <nitz.carola # googlemail.com>
+ *
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+import PassKit
+
+class DonationAmountView: UIStackView {
+
+    private let donationAmount = UITextField(frame: .zero)
+    private let donationTitle = UILabel(frame: .zero)
+    private let separator = UIView(frame: .zero)
+    private let separator2 = UIView(frame: .zero)
+    private var paymentButton: PKPaymentButton?
+    private var buttons: [UIButton]?
+    private let keyboard = UIButton(type: .custom)
+    let closebutton = UIButton(type: .custom)
+    private var donationManager: DonationManager?
+    
+    lazy var numberFormatter = { () -> NumberFormatter in
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .currency
+        formatter.locale = Locale.current
+        return formatter
+    }()
+
+    lazy var header: UIStackView = {
+        closebutton.setImage( UIImage(named: "close"), for: .normal)
+
+        donationTitle.font = .boldSystemFont(ofSize: 13)
+        donationTitle.text = NSLocalizedString("DONATE", comment: "")
+
+        let donationStackview = UIStackView(arrangedSubviews: [donationTitle, closebutton])
+        return donationStackview
+    }()
+
+    lazy var donationAmountView: UIStackView = {
+        let donation = UILabel(frame: .zero)
+        donation.font = .boldSystemFont(ofSize: 13)
+        donation.text = NSLocalizedString("DONATION_AMOUNT", comment: "")
+        donation.textColor = PresentationTheme.current.colors.cellDetailTextColor
+        donation.setContentHuggingPriority(.required, for: .horizontal)
+
+        donationAmount.font = .boldSystemFont(ofSize: 28)
+        donationAmount.text = numberFormatter.string(from: 5.0)
+        donationAmount.keyboardType = .decimalPad
+        donationAmount.delegate = self
+        donationAmount.textAlignment = .right
+
+        let donationStackview = UIStackView(arrangedSubviews: [donation, donationAmount])
+        donationStackview.alignment = .bottom
+        return donationStackview
+    }()
+
+    lazy var donationButtons: UIStackView = {
+        let ten = amountButton(amount: NSNumber(value: 10.0))
+        let twenty = amountButton(amount: NSNumber(value: 20.0))
+        let fifty = amountButton(amount:NSNumber(value: 50.0))
+        keyboard.addTarget(self, action: #selector(toggleKeyboard), for: .touchUpInside)
+        keyboard.sizeToFit()
+        keyboard.layer.cornerRadius = keyboard.frame.size.height / 2.0
+        keyboard.layer.borderWidth = 1.0
+        buttons = [ten, twenty, fifty, keyboard]
+        let donationStackview = UIStackView(arrangedSubviews: buttons!)
+        donationStackview.spacing = 16
+        donationStackview.distribution = .fillEqually
+        return donationStackview
+    }()
+
+
+    func setupPayButton() {
```

Comment:
Shouldn't this be a private method? Same for most of the `setup*()` methods in this PR.